File: Java/Xadrez.java
public class Xadrez {
    private String tipo;
    private String cor;
    private int[] posicao;

    public Xadrez(String tipo, String cor, int[] posicao) {
        this.tipo = tipo;
        this.cor = cor;
        this.posicao = posicao;
    }

    public void mover(int[] novaPosicao) {

    }

    public void capturar(Xadrez pecaOponente) {

    }

    public boolean ehMovimentoValido(int[] novaPosicao) {
        return false;
    }

    public String[] obterInformacoesPeca() {
        return new String[]{tipo, cor, posicao[0] + "," + posicao[1]};
    }
}

public class Rei extends Xadrez {
    public Rei(String cor, int[] posicao) {
        super("Rei", cor, posicao);
    }
}

public class Rainha extends Xadrez {
    public Rainha(String cor, int[] posicao) {
        super("Rainha", cor, posicao);
    }
}

public class TabuleiroXadrez {
    private Xadrez[][] tabuleiro;

    public TabuleiroXadrez() {
        this.tabuleiro = new Xadrez[8][8];
    }

    public boolean moverPeca(int[] posicaoAtual, int[] novaPosicao) {
        return false;
    }

    public boolean capturarPeca(int[] posicaoAtual, int[] posicaoOponente) {
        return false;
    }

    public boolean ehMovimentoValido(int[] posicaoAtual, int[] novaPosicao) {
        return false;
    }
}

File: C/Xadrez.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    char tipo[20];
    char cor[10];
    int posicao[2];
} Xadrez;

typedef struct {
    Xadrez **tabuleiro;
} TabuleiroXadrez;

Xadrez* criar_Xadrez(char tipo[], char cor[], int posicao[]);

void mover(Xadrez* peca, int novaPosicao[]);

void capturar(Xadrez* peca, Xadrez* pecaOponente);

int ehMovimentoValido(Xadrez* peca, int novaPosicao[]);

char** obterInformacoesPeca(Xadrez* peca);

TabuleiroXadrez* criar_TabuleiroXadrez();

int moverPeca(TabuleiroXadrez* tabuleiro, int posicaoAtual[], int novaPosicao[]);

int capturarPeca(TabuleiroXadrez* tabuleiro, int posicaoAtual[], int posicaoOponente[]);

int ehMovimentoValido(TabuleiroXadrez* tabuleiro, int posicaoAtual[], int novaPosicao[]);
